src/NuGet.Core/NuGet.PackageManagement/Projects/MSBuildNuGetProject.cs

```diff
@@ -618,9 +619,8 @@ private static bool IsAssemblyReference(string filePath)
                 return true;
             }
 
-            // Assembly reference must have a .dll|.exe|.winmd extension and is not a resource assembly;
-            return !filePath.EndsWith(Constants.ResourceAssemblyExtension, StringComparison.OrdinalIgnoreCase) &&
-                   Constants.AssemblyReferencesExtensions.Contains(Path.GetExtension(filePath), StringComparer.OrdinalIgnoreCase);
+            // Assembly reference must have a .dll|.exe|.winmd extension and is not a resource assembly:
```

Comment:
I have a concern  with this change because it's a behavior explicitly implemented in the past. 
Changing this by default could have some unintended consequences. 

Specifically my hesitation stems from the fact that the issue in question does not have a lot of customer engagement.

Reply:
Well, with a current change logic should remain the same. The previous implementation is filtering out all the assemblies that ended up with the 'resources' in a title that is not correct. 
Current filter is skipping only satellite assemblies under:
lib/<anything>/<anything>/*.resources.dll

The bug is reproduced by the unit test:
TestMSBuildNuGetProjectDontIgnoreResourceAssembly
Reverting the fix in the MSBuildNuGetProject.cs will clearly revial the bug.